contents/javascript/Javascript-promises.md

```diff
@@ -0,0 +1,198 @@
+# Javascript: Promises
+
+Author: Daniel Berzin Chua
+
+## Why Promises?
+
+We're used to operations completing their work before proceeding on with other operations in a sequential manner. These sorts of operations are synchronous, and they're usually easy to understand and debug. However, take a HTTP request for example. It is an operation that takes a while to process because there's a round trip time that involves the time for request to be sent to the server you are querying, and the time for the server to send the response back to you. It may take varying amounts of time depending on your internet speed and location.
```

Comment:
`an HTTP`? https://www.quora.com/A-http-or-an-http-which-is-the-correct-grammatical-usage